frame/treasury/src/lib.rs

```diff
@@ -321,13 +324,15 @@ pub mod pallet {
 		/// # </weight>
 		fn on_initialize(n: T::BlockNumber) -> Weight {
 			let pot = Self::pot();
-			let deactivated = Inactive::<T, I>::get();
+			let deactivated = Deactivated::<T, I>::get();
 			if pot != deactivated {
-				match (pot > deactivated, pot.max(deactivated) - pot.min(deactivated)) {
-					(true, delta) => T::Currency::deactivate(delta),
-					(false, delta) => T::Currency::reactivate(delta),
-				}
-				Inactive::<T, I>::put(&pot);
+				T::Currency::reactivate(deactivated);
+				T::Currency::deactivate(pot);
+				Deactivated::<T, I>::put(&pot);
```

Comment:
Also kill the `Inactive` here as cleanup?

Reply:
I'm actually unsure what the implications are for changing the logic here. Is it simply just reverting code?

Reply:
Its undoing and fixing the fund-deactivation since it was faultily implemented the first time.  
IIUC on the next run the `Deactivated` will be zero, so the whole pot will be `deactivate`d, which is fine since the foregoing balances migration will reset all deactivated funds.

Reply:
> Also kill the `Inactive` here as cleanup?

Let's delete it in a migration so that it happens once, not in `on_init`.

Reply:
> since it was faultily implemented the first time.

What's the case case where the old logic was wrong?

Reply:
Incorrect logic was in crowdloan where a datatype was being misused. See https://github.com/paritytech/polkadot/pull/6471